src/i_midipipe.c

```diff
@@ -215,6 +215,19 @@ void RemoveFileSpec(TCHAR *path, size_t size)
     *(fp + 1) = '\0';
 }
 
+static boolean BlockForAck(void)
+{
+    boolean ok;
+    net_packet_t *packet;
+
+    packet = NET_NewPacket(2);
+    NET_WriteInt16(packet, MIDIPIPE_PACKET_TYPE_REGISTER_SONG_ACK);
```

Comment:
This should be `MIDIPIPE_PACKET_TYPE_ACK`.  In fact, I would remove `MIDIPIPE_PACKET_TYPE_REGISTER_SONG_ACK` altogether.  Don't worry about breaking protocol compatibility, midiproc is aware of which version of Chocolate Doom is connecting and will refuse to communicate with mismatched versions.